tests/unit/lib/secrets/vault_csi_tests.py

```diff
@@ -0,0 +1,271 @@
+import datetime
+import json
+import shutil
+import string
+import tempfile
+import typing
+import unittest
+
+from pathlib import Path
+from unittest.mock import mock_open
+from unittest.mock import patch
+
+import gevent
+import pytest
+import typing_extensions
+
+from baseplate.lib.secrets import SecretNotFoundError
+from baseplate.lib.secrets import secrets_store_from_config
+from baseplate.lib.secrets import SecretsStore
+from baseplate.lib.secrets import VaultCSISecretsStore
+
+SecretType: typing_extensions.TypeAlias = typing.Dict[str, any]
+
+
+def write_secrets(secrets_data_path: Path, data: typing.Dict[str, SecretType]) -> None:
+    """Write secrets to the current data directory."""
+    for key, value in data.items():
+        secret_path = secrets_data_path.joinpath(key)
+        secret_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(secret_path, "w") as fp:
+            json.dump(value, fp)
+
+
+def write_symlinks(data_path: Path) -> None:
+    csi_path = data_path.parent
+    # This path can be monitored for changes
+    # https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/c697863c35d5431ec048b440d36550eb3ceb338f/pkg/util/fileutil/atomic_writer.go#L60-L62
+    data_link = Path(csi_path, "..data")
+    if data_link.exists():
+        # Simulate atomic update
+        new_data_link = Path(csi_path, "..data-new")
+        new_data_link.symlink_to(data_path)
+        new_data_link.rename(data_link)
+    else:
+        data_link.symlink_to(data_path)
```

Comment:
Any reason not to always do this?

```suggestion
    # Simulate atomic update
    new_data_link = Path(csi_path, "..data-new")
    new_data_link.symlink_to(data_path)
    new_data_link.rename(data_link)
```

Reply:
I tried several versions of this but I couldn't get it to work!